src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderExtensions.cs

```diff
@@ -278,11 +278,36 @@ private static bool IsTypeParameterEquivalentToTypeInst(this Type typeParam, Typ
 
         public static bool HasSameMetadataDefinitionAs(this MemberInfo mi1, MemberInfo mi2)
         {
+            if (s_lazyHasSameMetadataDefinitionAsApi == null)
+            {
+                MethodInfo apiMethod = typeof(MemberInfo).GetMethod(
```

Comment:
"Before the probing for MetadataToken" is becasue I think this would be more efficient if both are present.

---
In reply to: [121757009](https://github.com/dotnet/corefx/pull/20991#discussion_r121757009) [](ancestors = 121757009)